sklearn/decomposition/_fastica.py

```diff
@@ -574,12 +574,13 @@ def g(x, fun_args):
                     )
                 d[degenerate_idx] = eps  # For numerical issues
                 d = np.sqrt(d, d)
-                d, u = d[sort_indices], u[sort_indices]
-                # Resize and reorder to match svd
-                u = u[::-1, : min(X.shape) : -1]
+                d, u = d[sort_indices], u[:, sort_indices]
             else:
                 u, d = linalg.svd(XT, full_matrices=False, check_finite=False)[:2]
 
+            # Give consistent eigenvectors for both svd solvers
+            u *= np.sign(u[0])
```

Comment:
The alternative solution is:

```python
    max_abs_cols = np.argmax(np.abs(u), axis=0)
    signs = np.sign(u[max_abs_cols, range(u.shape[1])])
    u *= signs
```

If we want to be strictly the same as `svd_flip`.